src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -846,6 +847,40 @@ def clone_subvolume_snapshot(self, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    def _get_src_path(self, vol_handle, dst_group, dst_subvol):
+        src_subvol_details = dst_subvol._get_clone_source()
+        src_group_name = src_subvol_details.get('group', dst_group.NO_GROUP_NAME)
+        src_subvol_name = src_subvol_details['subvolume']
+
+        if src_group_name != dst_group.groupname:
+            with open_subvol_in_group(self.mgr, vol_handle, self.volspec,
+                                      src_group_name, src_subvol_name,
+                                      SubvolumeOpType.GETPATH) as src_subvol:
+                src_path = src_subvol.base_path.decode('utf-8')
+
+        else:
+            with open_subvol(self.mgr, vol_handle, self.volspec, dst_group,
+                             src_subvol_name, SubvolumeOpType.GETPATH) as \
+                             src_subvol:
+                src_path = src_subvol.base_path.decode('utf-8')
+
+        return src_path
+
+    def _get_clone_progress_report(self, vol_handle, dst_group, dst_subvol):
+        dst_path = dst_subvol.base_path.decode('utf-8')
+        src_path = self._get_src_path(vol_handle, dst_group, dst_subvol)
```

Comment:
Could we rename function name s/_get_src_path/_get_src_clone_path/ ?

Reply:
Fixed.